aten/src/ATen/native/Pow.cpp

```diff
@@ -31,10 +32,16 @@ Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
            "result type ", common_dtype, " can't be cast to the desired output type ",
            result.scalar_type());
 
-  if (exp.equal(0.0)) {
-    result.resize_as_(base).fill_(1);
-  } else if (exp.equal(1.0)) {
-    result.resize_as_(base).copy_(base);
+  auto exponent = (exp.isComplex()) ? exp.toComplexDouble() : exp.toDouble();
+
+  if (exponent == 0.0) {
```

Comment:
@Kiyosora can you please retain this check to `exp.equal(0.0)` (as in master)?

Reply:
Addressed!